### PR TITLE
Ota firmware updates

### PR DESCRIPTION
**Pull request by kYc0o**

### CODE REVIEW

File: sys/ota_update/ota_update.c [hunk 1]
```diff
@@ -0,0 +1,494 @@
+/*
```
Comment (kaspar030) [inline sys/ota_update/ota_update.c]: please split into tftp-server and coap+tftp-client

File: sys/include/ota_update.h [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+#define OTA_UPDATE_STACK          (THREAD_STACKSIZE_MAIN + THREAD_EXTRA_STACKSIZE_PRINTF + 4096)
+#endif
+
+/**
+ * @brief The message code to stop OTA client
+ */
+#define OTA_UPDATE_STOP           (0x1000)
+
+/**
+ * @brief Peiodic request time for update
```
Comment (kaspar030) [inline sys/include/ota_update.h]: periodic

File: boards/iotlab-common/Makefile.features [hunk 3]
```diff
@@ -11,5 +11,9 @@ FEATURES_PROVIDED += periph_uart
 # Various other features (if any)
 FEATURES_PROVIDED += cpp
 
+ifdef APP_ID
```
Comment (kaspar030) [inline boards/iotlab-common/Makefile.features]: please remove ifdef

File: sys/ota_update/ota_update.c (hunk 1 above)
Comment (kYc0o) [reply sys/ota_update/ota_update.c]: Done.

**Comment by kaspar030** [comment]
please rebase to master!

**Comment by kYc0o** [comment]
Done.

**Pull request closed without merging**